include/zephyr/drivers/usb/udc.h

```diff
@@ -76,6 +76,10 @@ struct udc_ep_caps {
 	uint32_t bulk : 1;
 	/** ISO transfer capable endpoint */
 	uint32_t iso : 1;
+	/** High-Bandwidth interrupt transfer capable endpoint */
+	uint32_t hb_interrupt : 1;
+	/** High-Bandwidth isochronous transfer capable endpoint */
+	uint32_t hb_iso : 1;
```

Comment:
I assume that if endpoint is capable of hb_interrupt, then it is capable of hb_iso. Is it not too cautious to have two flags?

Reply:
I wouldn't necessarily connect `hb_interrupt` with `hb_iso`. There are devices out there that support interrupt but do not support isochronous (and vice versa) on selected endpoints.

An alternative I was considering was to have `hb_capable` (or similar name) that would be a modifier. If `hb_capable` was set together with `interrupt` it would mean that endpoint can support both standard interrupt and high-bandwidth interrupt. Similarly `hb_capable` with `iso` would mean both standard iso and high-bandwidth iso.

I scrapped the modifier idea considering it is just a single bit in a bitfield... but if anything I would go with such modifier and not implying `hb_interrupt` means `hb_iso`.